src/client/components/pages/parts/search-results.js

```diff
@@ -21,122 +21,279 @@ import * as bootstrap from 'react-bootstrap';
+							}
+							{genEntityIconHTMLElement(result.type)}{_startCase(result.type)}
+						</td>
+					}
 					<td>
-						{genEntityIconHTMLElement(result.type)}{_startCase(result.type)}
+						<a href={link}>
+							{name} {disambiguation}
+						</a>
 					</td>
+					{
+						!this.props.condensed &&
+						<td>
+							{secondaryAliases}
+						</td>
+					}
+				</tr>
+			);
+		});
+		let tableCssClasses = 'table table-striped';
+		if (this.props.condensed) {
+			tableCssClasses += ' table-condensed';
+		}
+
+		return (
+			<div>
+				{
+					this.props.user ?
+						<div>
+							<AddToCollectionModal
+								bbids={this.state.selected.map(selected => selected.bbid)}
+								closeModalAndShowMessage={this.closeModalAndShowMessage}
+								entityType={this.state.selected[0]?.type}
+								handleCloseModal={this.onCloseModal}
+								show={this.state.showModal}
+								userId={this.props.user.id}
+							/>
+						</div> : null
 				}
-				<td>
-					<a href={link}>
-						{name} {disambiguation}
-					</a>
-				</td>
 				{
-					!props.condensed &&
-					<td>
-						{secondaryAliases}
-					</td>
+					!this.props.condensed &&
+					<h3 style={{color: '#754e37'}}>
+						Search Results
+					</h3>
 				}
-			</tr>
-		);
-	});
-	let tableCssClasses = 'table table-striped';
-	if (props.condensed) {
-		tableCssClasses += ' table-condensed';
-	}
-	return (
-		<div>
-			{
-				!props.condensed &&
-				<h3 style={{color: '#754e37'}}>
-					Search Results
-				</h3>
-			}
-			<hr className="thin"/>
-			<Table
-				responsive
-				className={tableCssClasses}
-			>
+				<hr className="thin"/>
+				<Table
+					responsive
+					className={tableCssClasses}
+				>
+					{
+						!this.props.condensed &&
+						<thead>
+							<tr>
+								<th className="col-sm-3">Type</th>
+								<th className="col-sm-5">Name</th>
+								<th className="col-sm-4">Aliases</th>
+							</tr>
+						</thead>
+					}
+					<tbody>
+						{results}
+					</tbody>
+				</Table>
 				{
-					!props.condensed &&
-					<thead>
-						<tr>
-							<th style={{width: '150px'}}>Type</th>
-							<th>Name</th>
-							<th>Aliases</th>
-						</tr>
-					</thead>
+					this.state.message.text ?
+						<Alert
+							bsStyle={this.state.message.type}
+							className="margin-top-1"
+							onDismiss={this.handleAlertDismiss}
+						>
+							{this.state.message.text}
+						</Alert> : null
+
+				}
+				{
+					this.props.user ?
+						<ButtonGroup>
+							<Button
+								bsStyle="primary"
+								type="button"
+								onClick={this.handleAddToCollection}
+							>
+								<FontAwesomeIcon icon="grip-vertical"/>
+								&nbsp;Add to Collection
+							</Button>
+							{
+								this.state.selected.length ?
+									<Button
+										bsStyle="danger"
+										type="button"
+										onClick={this.handleClearSelected}
+									>
+										Clear <span className="circle-number">{this.state.selected.length}</span> selected
+									</Button> : null
+							}
```

Comment:
A couple of suggestions for this button:
- using the `<Badge>` element for Bootstrap for the number
- making the button disabled instead of removing it if there is no selection (i'm of the opinion we should do the same for the "add to collection" button)
- changing the color (to keep red for more important destructive actions)

```suggestion
							<Button
								bsStyle="warning"
								disabled={!this.state.selected.length}
								type="button"
								onClick={this.handleClearSelected}
							>
								Clear <Badge>{this.state.selected.length}</Badge> selected
							</Button>
```